python/ray/experimental/gpu_object_manager/gpu_object_store.py

```diff
@@ -82,40 +83,96 @@ def __ray_recv__(
         tensor = torch.zeros(shape, dtype=dtype, device=device)
         collective.recv(tensor, src_rank, group_name=communicator_name)
         tensors.append(tensor)
-    gpu_object_store.add_gpu_object(obj_id, tensors)
+    gpu_object_store.add_object(obj_id, tensors)
+
+
+def __ray_get_tensor_meta__(self, obj_id: str):
+    """Helper function that runs on the src actor to get the tensor metadata."""
+    from ray._private.worker import global_worker
+
+    gpu_object_store = global_worker.gpu_object_manager.gpu_object_store
+    # NOTE: We do not specify a timeout here because the user task that returns
+    # it could take arbitrarily long and we don't want to trigger a spurious
+    # timeout.
+    gpu_object_store.wait_object(obj_id)
+    tensors = gpu_object_store.get_object(obj_id)
+    return [(t.shape, t.dtype) for t in tensors]
 
 
 def __ray_fetch_gpu_object__(self, obj_id: str):
     """Helper function that runs on the src actor to fetch tensors from the GPU object store via the object store."""
     from ray._private.worker import global_worker
 
     gpu_object_store = global_worker.gpu_object_manager.gpu_object_store
-    assert gpu_object_store.has_gpu_object(
+    assert gpu_object_store.has_object(
         obj_id
     ), f"obj_id={obj_id} not found in GPU object store"
-    tensors = gpu_object_store.get_gpu_object(obj_id)
+    tensors = gpu_object_store.get_object(obj_id)
     # TODO(kevin85421): The current garbage collection implementation for the
```

Comment:
`return gpu_object_store.pop_object(obj_id)`?

Reply:
Ah I think this will need to be removed anyway after your GC PR.